notmuch/private.h

```diff
@@ -34,11 +34,15 @@ struct Mailbox;
 #endif
 
 /* The definition in <notmuch.h> is broken */
-#define LIBNOTMUCH_CHECK_VERSION(major, minor, micro)                             \
-  (LIBNOTMUCH_MAJOR_VERSION > (major) ||                                          \
-   (LIBNOTMUCH_MAJOR_VERSION == (major) && LIBNOTMUCH_MINOR_VERSION > (minor)) || \
-   (LIBNOTMUCH_MAJOR_VERSION == (major) &&                                        \
-    LIBNOTMUCH_MINOR_VERSION == (minor) && LIBNOTMUCH_MICRO_VERSION >= (micro)))
+// notmuch 0.32 introduced libnotmuch 5.4, but didn't include version bump until 0.32.3
+// notmuch 0.26 introduced libnotmuch 5.1, but didn't include version bump until 0.26.1
+#define LIBNOTMUCH_CHECK_VERSION(major, minor, micro)                              \
+  (major == 5 && minor == 4 && defined(HAVE_NOTMUCH_DATABASE_OPEN_WITH_CONFIG)) || \
+  (major == 5 && minor == 1 && defined(HAVE_NOTMUCH_DATABASE_INDEX_FILE))       || \
+  ((LIBNOTMUCH_MAJOR_VERSION > (major) ||                                          \
+   (LIBNOTMUCH_MAJOR_VERSION == (major) && LIBNOTMUCH_MINOR_VERSION > (minor))  || \
+   (LIBNOTMUCH_MAJOR_VERSION == (major) &&                                         \
+    LIBNOTMUCH_MINOR_VERSION == (minor) && LIBNOTMUCH_MICRO_VERSION >= (micro))))
```

Comment:
The `defined()` preprocessor directives, in a `#define` are causing build problems:

- gcc 11.2.1: `notmuch/db.c:108:37: error: this use of "defined" may not be portable`
- clang 12.0.1: `notmuch/db.c:108:5: error: macro expansion producing 'defined' has undefined behavior`

Perhaps:
```c
#ifndef HAVE_NOTMUCH_DATABASE_INDEX_FILE
#define HAVE_NOTMUCH_DATABASE_INDEX_FILE 0
#endif

#ifndef HAVE_NOTMUCH_DATABASE_OPEN_WITH_CONFIG
#define HAVE_NOTMUCH_DATABASE_OPEN_WITH_CONFIG 0
#endif

#define LIBNOTMUCH_CHECK_VERSION(major, minor, micro)                              \
  (major == 5 && minor == 4 && HAVE_NOTMUCH_DATABASE_OPEN_WITH_CONFIG) ||          \
  (major == 5 && minor == 1 && HAVE_NOTMUCH_DATABASE_INDEX_FILE)       ||          \
  ((LIBNOTMUCH_MAJOR_VERSION > (major) ||                                          \
   (LIBNOTMUCH_MAJOR_VERSION == (major) && LIBNOTMUCH_MINOR_VERSION > (minor))  || \
   (LIBNOTMUCH_MAJOR_VERSION == (major) &&                                         \
    LIBNOTMUCH_MINOR_VERSION == (minor) && LIBNOTMUCH_MICRO_VERSION >= (micro))))
```

I'll keep thinking

Reply:
Incorporated change